core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParserErrorHandling.java

```diff
@@ -160,7 +160,9 @@ private static Stream<Arguments> statements()
                 Arguments.of("SELECT * FROM t FOR TIMESTAMP AS OF TIMESTAMP WHERE",
                         "line 1:52: mismatched input '<EOF>'. Expecting: <expression>"),
                 Arguments.of("SELECT * FROM t FOR VERSION AS OF TIMESTAMP WHERE",
-                        "line 1:50: mismatched input '<EOF>'. Expecting: <expression>"));
+                        "line 1:50: mismatched input '<EOF>'. Expecting: <expression>"),
+                Arguments.of("SELECT ROW(DATE '2022-10-10', DOUBLE 12.0)",
```

Comment:
I think I found a simpler repro:
```
VALUES(DATE 2);
```
The error message is incorrect, as it says `line 1:13: mismatched input 'DATE'`.

However, when we skip the parentheses, the error message is correct:
```
VALUES DATE 2;
```
It says `line 1:13: mismatched input '2'`. 

Maybe you could add these examples, or similar ones, as test cases, or include them in the commit message? I think it would be good to have some more explanation of what this change does in practice.

Reply:
Another repro:
```
SELECT count(DISTINCT *) FROM (VALUES 1);
Query 20230109_124153_00008_e286n failed: line 1:23: mismatched input 'count'. Expecting: '*', <expression>
```